liquibase-maven-plugin/src/main/java/org/liquibase/maven/plugins/LiquibaseMojo.java

```diff
@@ -0,0 +1,163 @@
+package org.liquibase.maven.plugins;
+
+import liquibase.Scope;
+import liquibase.command.CommandScope;
+import liquibase.configuration.AbstractMapConfigurationValueProvider;
+import liquibase.configuration.LiquibaseConfiguration;
+import liquibase.integration.commandline.Main;
+import org.apache.maven.execution.MavenSession;
+import org.apache.maven.plugin.AbstractMojo;
+import org.apache.maven.plugin.MojoExecution;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugin.MojoFailureException;
+import org.apache.maven.project.MavenProject;
+import org.codehaus.plexus.component.MapOrientedComponent;
+import org.codehaus.plexus.component.configurator.ComponentConfigurationException;
+import org.codehaus.plexus.component.repository.ComponentRequirement;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+
+public class LiquibaseMojo extends AbstractMojo implements MapOrientedComponent  {
+
+    private Map<String, Object> configuration;
+    private MavenProject project;
+    private MojoExecution mojoExecution;
+
+    public LiquibaseMojo() {
+    }
+
+    @Override
+    public void addComponentRequirement(ComponentRequirement componentRequirement, Object o) throws ComponentConfigurationException {
+        //TODO: How does this get used?
```

Comment:
Is this important?

Reply:
This was probably a comment I put in as I was trying to figure out how the plugin system works